services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -519,17 +519,21 @@ specDeleteUser = do
                 !!! const 405 === statusCode
 
     describe "DELETE /Users/:id" $ do
-        it "when called twice, should first delete then 404 you" $ do
+        it "should respond with 204" $ do
             (tok, _) <- registerIdPAndScimToken
             user <- randomScimUser
             storedUser <- createUser tok user
             let uid = scimUserId storedUser
 
             spar <- view teSpar
+            -- Expect first call to succeed
             deleteUser_ (Just tok) (Just uid) spar
                 !!! const 204 === statusCode
+            -- The second call may return either of 204 or 404 depending on whether Brig has
+            -- finished deletion. This assertion is here to document that this is currently
+            -- the expected behaviour
             deleteUser_ (Just tok) (Just uid) spar
-                !!! const 404 === statusCode  -- https://tools.ietf.org/html/rfc7644#section-3.6
```

Comment:
I think this should instead wait until a 404 actually happens... otherwise how do we ensure that users actually get deleted?

Reply:
Hrmm this is a good point; do you have a suggestion of how to do this cleanly? I want to avoid having a bunch of `threadDelay`s slowing down our tests

Reply:
we could use the same retry trick we use elsewhere.  we just keep deleting until we get a 404.

grep for `retry` or `Control.Retry`.

Reply:
(should have thought of that when you asked me, sorry!)